src/control/server/mgmt_pool.go

```diff
@@ -397,20 +395,24 @@ func (svc *mgmtSvc) PoolDestroy(ctx context.Context, req *mgmtpb.PoolDestroyReq)
 
 	svc.log.Debugf("MgmtSvc.PoolDestroy dispatch, resp:%+v\n", resp)
 
-	switch drpc.DaosStatus(resp.Status) {
-	case drpc.DaosSuccess:
+	ds := drpc.DaosStatus(resp.Status)
+	switch ds {
+	case drpc.DaosSuccess, drpc.DaosNotLeader, drpc.DaosNotReplica:
+		if ds == drpc.DaosNotLeader || ds == drpc.DaosNotReplica {
+			// If we're not cleaning up, then this is an error.
+			if len(req.SvcRanks) != len(ps.Storage.CreationRanks()) {
```

Comment:
can you explain why the service rank list should be the same as the storage rank list in this case? I don't fully understand

Reply:
I think it might be better to abstract that away. req.SvcRanks is set above on lines 377 and 383, and would be set to include all storage ranks if the pool was already requested to be destroyed.

Reply:
Yeah, I had a standalone variable indicating that we were in cleanup mode but at the time it seemed neater to do it this way. I'll rework it to be clear, as it's particularly important that the logic here should be understandable.